beacon-chain/blockchain/chain_info.go

```diff
@@ -274,6 +277,43 @@ func (s *Service) HeadETH1Data() *ethpb.Eth1Data {
 	return s.head.state.Eth1Data()
 }
 
+// HeadValidatorAtIndex returns the validator at the given index in the head state.
+// Error is returned if head state is nil.
+func (s *Service) HeadValidatorAtIndex(ctx context.Context, index primitives.ValidatorIndex) (state.ReadOnlyValidator, error) {
+	s.headLock.RLock()
+	defer s.headLock.RUnlock()
+
+	if !s.hasHeadState() {
+		return nil, ErrNilHead
+	}
+
+	return s.head.state.ValidatorAtIndexReadOnly(index)
+}
+
+// HeadGenesisRoot returns the genesis root of the current head state.
+func (s *Service) HeadGenesisRoot() [32]byte {
+	s.headLock.RLock()
+	defer s.headLock.RUnlock()
+
+	if !s.hasHeadState() {
+		return [32]byte{}
+	}
+	return [32]byte(s.head.state.GenesisValidatorsRoot())
```

Comment:
why isn' t this hardcoded by now.

Reply:
We ask each other this every six months. I'll open an issue and see if anyone can take it
https://github.com/prysmaticlabs/prysm/issues/14364